.github/workflows/integration_test_charm.yaml

```diff
@@ -182,6 +182,7 @@ jobs:
           juju add-model test
           pipx install tox
           pipx install poetry
+          pipx install git+https://github.com/PietroPasotti/jhack.git
```

Comment:
is it a good idea to install an unpinned  jhack from someone's private github repo? how will this be considered by IS when running on self-hosted runners?

Reply:
@carlcsaposs-canonical true, but you insisted on pinning a specific commit, otherwise we are in trouble if Pietro GH account has hacked. :-)

Reply:
I see it different. Canonical org repos are managed in a centralized way.
IS (is/will) check(ing) all of them periodically and enforce new security measures.

Private repos are private. Developers are coming and leaving, people are also dying.
I do trust Pietro. He is superb! However rules are general for the reason: if workflow uses 3rd-party repo - pin the commit you have checked and going to use, IMHO.